ibverbs/src/lib.rs

```diff
@@ -533,6 +553,97 @@ impl<'ctx> CompletionQueue<'ctx> {
             Ok(&mut completions[0..n as usize])
         }
     }
+
+    /// Waits for one or more work completions in a Completion Queue (CQ).
+    ///
+    /// Unlike `poll`, this method blocks until at least one work completion is available or the
+    /// optional timeout expires. It is designed to wait efficiently for completions when polling
+    /// alone is insufficient, such as in low-traffic scenarios.
+    ///
+    /// The returned slice reflects completed work requests (e.g., sends, receives) from the
+    /// associated Work Queue. Not all fields in `ibv_wc` are valid unless the status is
+    /// `IBV_WC_SUCCESS`.
+    ///
+    /// # Errors
+    /// - `TimedOut`: If the timeout expires before any completions are available.
+    /// - System errors: From underlying calls like `req_notify_cq`, `poll`, or `ibv_get_cq_event`.
+    pub fn wait<'c>(
+        &self,
+        completions: &'c mut [ffi::ibv_wc],
+        timeout: Option<Duration>,
+    ) -> io::Result<&'c mut [ffi::ibv_wc]> {
+        let c = completions as *mut [ffi::ibv_wc];
+
+        //
+        loop {
+            let polled_completions = self.poll(unsafe { &mut *c })?;
```

Comment:
Instead of having the `unsafe` here, can't you just use `completions` directly and only cast it to `c` until below this?

Reply:
Ah yes, this is https://github.com/rust-lang/rust/issues/54663 :disappointed: Worth adding that to a comment for why this needs to be `unsafe` then.